src/GenQPSolver.h

```diff
@@ -82,6 +82,21 @@ class TASKS_DLLAPI GenQPSolver
 	virtual void updateSize(int nrVars, int nrEq, int nrInEq, int nrGenInEq) = 0;
 
 	/**
+	* Setup dependent variables, only linear dependency are supported
+	* @param fullToReduced A full to reduced variables vector
+	* @param reducedToFull A reduced to full variables vector
+	* @param dependencies List of tuple {primary, replica, factor}
+	*/
+	virtual void setReductionParameters(std::vector<int> fullToReduced,
+																			std::vector<int> reducedToFull,
+																			std::vector<std::tuple<int, int, double>> dependencies)
+	{
+		fullToReduced_ = fullToReduced;
```

Comment:
See my comment down below, but unless we build `fullToReduced` from `dependencies` in this function, sanity checks should be added, either in the form of a `sSetReductionParameters` or debug-only asserts.